### PR TITLE
MWPW-158147-Eager load hero marque foreground image

### PR DESCRIPTION
**Pull request by Ruchika4**
* Eager load hero marque foreground image

Resolves: [MWPW-158147](https://jira.corp.adobe.com/browse/MWPW-158147)

**Test URLs:**
- Before: https://stage--cc--adobecom.hlx.live/drafts/ruchika/performance/illustrator?martech=off
- After: https://hero-marquee--cc--adobecom.hlx.live/drafts/ruchika/performance/illustrator?martech=off

### CODE REVIEW

File: creativecloud/scripts/utils.js [hunk 1]
```diff
@@ -62,6 +62,19 @@ function defineDeviceByScreenSize() {
   return 'TABLET';
 }
 
+function heroForegroundImage(firstBlock) {
+  let rows = firstBlock.querySelectorAll(':scope > div');
+  if (rows.length > 1 && rows[0].textContent !== '') {
```
Comment (aishwaryamathuria) [inline creativecloud/scripts/utils.js:67]: if (rows.length > 1 && rows[0].textContent !== '') rows.shift()

File: creativecloud/scripts/utils.js [hunk 2]
```diff
@@ -62,6 +62,19 @@ function defineDeviceByScreenSize() {
   return 'TABLET';
 }
 
+function heroForegroundImage(firstBlock) {
+  let rows = firstBlock.querySelectorAll(':scope > div');
```
Comment (aishwaryamathuria) [inline creativecloud/scripts/utils.js:66]:
```suggestion
  let rows = [...firstBlock.querySelectorAll(':scope > div')];
```

File: creativecloud/scripts/utils.js (hunk 1 above)
Comment (aishwaryamathuria) [inline creativecloud/scripts/utils.js:67]:
```suggestion
  if (rows.length > 1 && rows[0].textContent !== '') rows.shift();
```

File: creativecloud/scripts/utils.js [hunk 3]
```diff
@@ -62,6 +62,19 @@ function defineDeviceByScreenSize() {
...
+  if (rows.length > 1 && rows[0].textContent !== '') {
+    const [, ...tail] = rows;
+    rows = tail;
+  }
```
Comment (aishwaryamathuria) [inline creativecloud/scripts/utils.js:70]: we can remove const [, ...tail] = rows;
    rows = tail;
  }

File: creativecloud/scripts/utils.js [hunk 4]
```diff
@@ -62,6 +62,19 @@ function defineDeviceByScreenSize() {
...
+  const mainRowIndex = rows.findIndex((row) => {
+    const firstColText = row.children[0].textContent.toLowerCase().trim();
+    return !firstColText.includes('con-block-row-');
+  });
+  return rows[mainRowIndex];
```
Comment (aishwaryamathuria) [inline creativecloud/scripts/utils.js:75]: what if foreground image is not authored?

File: creativecloud/scripts/utils.js [hunk 5]
```diff
@@ -109,7 +122,11 @@ function getDecorateAreaFn() {
         if (bgImages?.querySelectorAll('img').length === 1 && bgImages.querySelectorAll('div').length === 1) eagerLoad(bgImages?.querySelector('div img'));
         else eagerLoad(bgImages?.querySelector(`:scope ${lcpImgVP[viewport]}`));
         // Foreground image
-        eagerLoad(firstBlock.querySelector(':scope div:last-child > div img'));
+        if (firstBlock?.classList.contains('hero-marquee')) {
+          const foreground = heroForegroundImage(firstBlock);
+          const imageHidden = (viewport === 'TABLET' && firstBlock?.classList.contains('media-hidden-tablet')) || (viewport === 'MOBILE' && firstBlock?.classList.contains('media-hidden-mobile'));
+          if (!imageHidden) eagerLoad(foreground.querySelector('img'));
```
Comment (aishwaryamathuria) [inline creativecloud/scripts/utils.js:128]: If foreground is empty this would error out

**Review by aishwaryamathuria — CHANGES_REQUESTED** [review]

File: creativecloud/scripts/utils.js (hunk 1 above)
Comment (Ruchika4) [reply creativecloud/scripts/utils.js]: done

File: creativecloud/scripts/utils.js (hunk 3 above)
Comment (Ruchika4) [reply creativecloud/scripts/utils.js]: I don't this is needed as shift() will remove the head row.

**Review by ryanmparrish — APPROVED** [review]

**Review by ryanmparrish — APPROVED** [review]

**Review by aishwaryamathuria — APPROVED** [review]

**Comment by spadmasa** [comment]
Validations done in below url 
https://hero-marquee--cc--adobecom.hlx.live/drafts/ruchika/performance/illustrator?martech=off
{B680C06D-B1CB-4F40-BB1D-98B1B0117775}
{BFBD9F13-A4CD-4120-B752-AA30E04AF598}

{08FE2F29-CF34-4C88-825E-C9CC243085B5}
details of validation is updated in https://jira.corp.adobe.com/browse/MWPW-158147
cc : @sigadamvenkata

**Review by suhjainadobe — APPROVED** [review]

**Review by salonijain3 — APPROVED** [review]

**Pull request merged**